feat(curriculum): dice game project (#50819)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/649f0fb16b7e27027526f4da.md

---
id: 649f0fb16b7e27027526f4da
title: Step 16
challengeType: 0
dashedName: step-16
---

# --description--

> Add a `if` statement where if `rolls` equates to `3`, it will send an `window.alert` message saying `You have made three rolls this round. Please select a score.`

Reviewer: ```suggestion
Add an `if` statement where if `rolls` equates to `3`, it will send a `window.alert` message saying `You have made three rolls this round. Please select a score.`
```

Author: done

> Add an `if` statement where if `rolls` equates to `3`, it will send a `window.alert` message saying `You have made three rolls this round. Please select a score.`

Reviewer 2: ```suggestion
Use an `if` condition to check if `rolls` is strictly equal to the number `3`. When the condition is true, use the `window.alert()` method to display the message `You have made three rolls this round. Please select a score.`.
```

# --hints--

> You should add a `if` statement where the condition is `rolls` equates to `3`.

Reviewer: ```suggestion
You should add a `if` statement where the condition is `rolls` equates to `3`. Use the strict equality ( `===` ) operator.
```

Author: done

```js
assert.match(code, /if\s*\(\s*rolls\s*===\s*3\s*\)\s*{/);
```

> After the condition is met, you should add a `window.alert` message saying `You have made three rolls this round. Please select a score.`

Reviewer 2: This should be two tests. One to check that `window.alert` is called, and the second to check the arugment passed is correct.

```js

> assert.match(code, /if\s*\(\s*rolls\s*===\s*3\s*\)\s*{\s*window\.alert\s*\(\s*['"]\s*You\s+have\s+made\s+three\s+rolls\s+this\s+round\.\s+Please\s+select\s+a\s+score\.\s*['"]\s*\)\s*}/);

Author: done
